primap2/_downscale.py

```diff
@@ -11,12 +11,12 @@ class DataArrayDownscalingAccessor(BaseDataArrayAccessor):
     def downscale_timeseries(
         self,
         *,
-        dim: str,
-        basket: str,
-        basket_contents: Sequence[str],
+        dim: Hashable,
```

Comment:
What does changing the type from str to Hashable mean in practice?

Reply:
The main reason to do it is that all the xarray-Functions take `Hashable`, so it makes mypy happy. In theory, it is possible to use any hashable type (for example, an int, or a tuple of strings) as a dimension name for xarray. In practice, we should never ever use anything else than a str, I think. I tried writing a test function with an int dimension name, and something inside xarray breaks before primap2 can even start complaining.

I liked `str` more because it is more explicit and we don't support other Hashables, but even though `str`s are a subset of Hashables, somehow static type inspection is only happy with `Hashable`. :confused:

Reply:
OK, got it